middle_end/flambda2/terms/flambda_primitive.ml

```diff
@@ -1295,10 +1317,11 @@ type binary_primitive =
   | Bigarray_get_alignment of int
   | Atomic_exchange
   | Atomic_fetch_and_add
+  | Array_vector_load of Lambda.boxed_vector * Array_kind.t * Mutability.t
```

Comment:
Why is this a separate primitive instead of having the `Array_load` primitive take a `array_accessor_width` payload similar to the string/bigstring case ?

In any case, it seems that having both the `boxed_vector` and the `Array_kind.t` is redundant (and possibly dangerous if/when the two disagree), and since the arrya kind can be computed from the `boxed_vector` I'd say to only keep that one.

Reply:
Now merged into `Array_load` / `Array_set`